frontend/src/components/Signup.tsx

```diff
@@ -0,0 +1,194 @@
+'use client';
+import { useAuthStore, useUserStore} from '../context/authStore';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { FormEvent, useState } from 'react';
+import {
+  QueryClient,
+  QueryClientProvider,
+  useMutation
+} from '@tanstack/react-query'
+
+const queryClient = new QueryClient();
+
+export default function Signup() {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <SignupForm />
+    </QueryClientProvider>
+  );
+}
+
+function SignupForm() {
+  const [formData, setFormData] = useState({ username: '', email: '', password: '' });
+  const [error, setError] = useState('');
+  const [loading, setIsLoading] = useState(false);
+
+  const { setIsLoggedIn } = useAuthStore();
+  const { setUser } = useUserStore();
+
+  const router = useRouter();
+
+  const mutation = useMutation({
```

Comment:
This method is similar in both sign up & log in.  Create a common method for it.